lib/core-integration/src/Test/Integration/Framework/DSL.hs

```diff
@@ -762,6 +766,27 @@ syncProgress =
     _set :: HasType (ApiT SyncProgress) s => (s, SyncProgress) -> s
     _set (s, v) = set typed (ApiT v) s
 
+metrics :: HasType StakePoolMetrics s => Lens' s StakePoolMetrics
+metrics =
+    lens _get _set
+  where
+    _get :: HasType StakePoolMetrics s => s -> StakePoolMetrics
+    _get = view typed
+    _set :: HasType StakePoolMetrics s => (s, StakePoolMetrics) -> s
+    _set (s, v) = set typed v s
+
+stake
+    :: HasType (Quantity "lovelace" Natural) s
+    => Lens' s (Quantity "lovelace" Natural)
+stake =
+    lens (view typed) (\(s, v) -> set typed v s)
+
+blocks
+    :: HasType (Quantity "block" Natural) s
+    => Lens' s (Quantity "block" Natural)
+blocks =
+    lens (view typed) (\(s, v) -> set typed v s)
```

Comment:
These lenses are somewhat pointless here :grimacing: ... The other lenses in this file are usually dealing with type conversions to actually makes it easier to write assertions after. For instance, this one would be better defined as:

```hs
blocks
    :: HasType (Quantity "block" Natural) s
    => Lens' s Natural
```

Reply:
That specifically doesn't work because it complains about overlapping instances. I believe because of the following, already existing, definition:
```haskell
amount :: HasType (Quantity "lovelace" Natural) s => Lens' s Natural
```

So I wrote it using `HasField'` instead:
```haskell
stake
    :: HasField' "controlledStake" s (Quantity "lovelace" Natural)
    => Lens' s Natural
stake = lens
    (getQuantity . getField @"controlledStake")
    (\(s, v) -> setField @"controlledStake" (Quantity v) s)

blocks
    :: HasField' "producedBlocks" s (Quantity "block" Natural)
    => Lens' s Natural
blocks = lens
    (getQuantity . getField @"producedBlocks")
    (\(s, v) -> setField @"producedBlocks" (Quantity v) s)
```